modules/EPMC.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+
+import argparse
+import sys
+import logging
+
+import pyspark
+from pyspark.sql.types import StringType
+from pyspark.sql import SparkSession 
+import pyspark.sql.functions as pf
+from pyspark import SparkContext
+from pyspark.conf import SparkConf
+
+
+# The following target labels are excluded as they were grounded to too many target Ids
+EXCLUDED_TARGET_TERMS = ['TEC', 'TECS', 'Tec', 'tec', '\'', '(', ')', '-', '-S', 'S', 'S-', 'SS', 'SSS',
+         'Ss', 'Ss-', 's', 's-', 'ss', 'U3', 'U6', 'u6', 'SNORA70', 'U2', 'U8']
+
+def main(cooccurrenceFile, outputFile, local=False):
+
+    ##
+    ## Initialize spark session
+    ##
+    if local:
+        sparkConf = (SparkConf()
+                 .set('spark.driver.memory', '15g')
+                 .set('spark.executor.memory', '15g')
+                 .set('spark.driver.maxResultSize', '0')
+                 .set('spark.debug.maxToStringFields', '2000')
+                 .set('spark.sql.execution.arrow.maxRecordsPerBatch', '500000')
+                 )
+        spark = (
+            SparkSession.builder
+                .config(conf=sparkConf)
+                .master('local[*]')
+                .getOrCreate()
+        )
+    else:
+        sparkConf = (SparkConf()
+                 .set('spark.driver.maxResultSize', '0')
+                 .set('spark.debug.maxToStringFields', '2000')
+                 .set('spark.sql.execution.arrow.maxRecordsPerBatch', '500000')
+                 )
+        spark = (
+            SparkSession.builder
+                .config(conf=sparkConf)
+                .getOrCreate()
+        )
+
+    logging.info(f'Spark version: {spark.version}')
+
+    ##
+    ## Log parameters:
+    ##
+    logging.info(f'Cooccurrence file: {cooccurrenceFile}')
+    logging.info(f'Output file: {outputFile}')
+    logging.info(f'Generating evidence:')
+
+    ##
+    ## Load/filter datasets:
+    ##
+    filtered_cooccurrence_df = (
+        # Reading file:
+        spark.read.parquet(cooccurrenceFile)
+
+        # Filtering for disease/target cooccurrences:
+        .filter(
+            (pf.col('type') == 'GP-DS') &  # Filter gene/protein - disease cooccurrence
+            (pf.col('isMapped') == True) &  # Filtering for mapped cooccurrences
+            (pf.col('pmid').isNotNull()) &  # Excluding publications without pmid
+            (pf.length(pf.col('text')) < 600) & # Exclude sentences with more than 600 characters
+            (pf.col('label1').isin(EXCLUDED_TARGET_TERMS) == False) # Excluding target labels from the exclusion list
```

Comment:
PEP8 strikes again; [it wants at least two spaces](https://www.python.org/dev/peps/pep-0008/#inline-comments) between a statement and a comment

```suggestion
            (pf.length(pf.col('text')) < 600) &  # Exclude sentences with more than 600 characters
            (pf.col('label1').isin(EXCLUDED_TARGET_TERMS) == False)  # Excluding target labels from the exclusion list
```